Pass Docker tags as build-args to dockerfile (#6733)
kyma-project/test-infra: development/image-builder/README.md

Currently, image-builder contains a basic implementation of a notary signer. If you want to add a new signer, refer to the [`sign`](./sign) package, and its code.

## Named Tags

> image-builder supports passing name along with tag both via -tag option or config for tag template.

Reviewer: ```suggestion
image-builder supports passing the name along with the tag both using the `-tag` option or config for the tag template.
```

> You can use `-tag name=value` to pass name for tag. 

Reviewer: ```suggestion
You can use `-tag name=value` to pass the name for the tag. 
```

> If name is not provided, it will be evaluated from value:
>  - if value is string it will be used as name directly eg.: 
>    - `-tag latest` will be equal to `-tag latest=latest`
>  - if value is go-template it will be converted to valid name eg:
>    - `-tag v{{ .ShortSHA }}-{{ .Date }}` will be equal to `-tag vShortSHA-Date=v{{ .ShortSHA }}-{{ .Date }}`

Reviewer: ```suggestion
If the name is not provided, it is evaluated from the value:
 - if the value is a string, it is used as a name directly. For example,`-tag latest` is equal to `-tag latest=latest`
 - if the value is go-template, it will be converted to a valid name. For example, `-tag v{{ .ShortSHA }}-{{ .Date }}` is equal to `-tag vShortSHA-Date=v{{ .ShortSHA }}-{{ .Date }}`
```

        Additional tag that the image will be tagged
  -variant string
        If variants.yaml file is present, define which variant should be built. If variants.yaml is not present, this flag will be ignored

>   -export-tags

Reviewer 2: I cannot find a test-case for this flag. Can you include it in the tests?

Author: Sure, I will do that shortly

>         Export parsed tags as build-args into dockerfile. Each tag will have number added to it in format `DOCKER_TAG_x` where x is index of tags array.

Reviewer: ```suggestion
        Export parsed tags as build-args into Dockerfile. Each tag will have a number added to it in format `DOCKER_TAG_x` where x is index of tags array.
```

  -silent
        Do not push build logs to stdout
  -tag value

>         Additional tag that the image will be tagged. Optionaly you can pass name in format name=value which will be used by export-tags.

Reviewer: ```suggestion
        Additional tag that the image will be tagged with. Optionally you can pass the name in the format name=value which will be used by export-tags.
```

  -variant string
        If variants.yaml file is present, define which variant should be built. If variants.yaml is not present, this flag will be ignored
  -export-tags

>         Export parsed tags as build-args into Dockerfile. Each tag will have format TAG_x where x is either tag name passed along with tag (see: Named Tags section). 

Reviewer: ```suggestion
        Export parsed tags as build-args into Dockerfile. Each tag will have format TAG_x, where x is the tag name passed along with the tag (see: Named Tags section). 
```

        Additional tag that the image will be tagged
  -variant string
        If variants.yaml file is present, define which variant should be built. If variants.yaml is not present, this flag will be ignored
  -export-tags

>         Export parsed tags as build-args into dockerfile. Each tag will have format TAG_x where x is either tag name passed along with tag (see: Named Tags section). 

Reviewer: ```suggestion
        Export parsed tags as build-args into Dockerfile. Each tag will have format TAG_x where x is either tag name passed along with tag (see: Named Tags section). 
```
